File: (6 kyu) Decode the Morse code/(6 kyu) Decode the Morse code.js
decodeMorse = function(morseCode) {
  return morseCode
    .trim() //remove extra spaces
    .split('   ') //devide by words
    .map((
      str, //parse each word
    ) =>
      str
        .split(' ') //devide by symbols
        .map((s) => MORSE_CODE[s]) //return Char
        .join(''),
    )
    .join(' ');
};

File: 6kyu/(6 kyu) Decode the Morse code/(6 kyu) Decode the Morse code.py
def decodeMorse(morse_code):
    result = []
    for word in morse_code.strip().split("   "):
        decoded_word = ""
        for letter in word.split():
            decoded_word += MORSE_CODE[letter]
        result.append(decoded_word)
    return " ".join(result)
